### DISCUSSION TITLE
I wanted to make a custom blockchain module to fetch data using external api and then store or edit in blockchain

### DISCUSSION THREAD

**Discussion by prakharadmybrand** [cosmos/community#23] [Chain Builders (Cosmos SDK)]
**I wanted to make a custom blockchain module to fetch data using external api and then store or edit in blockchain**. For that i was using ignite cli. I found a github repository that implemented the same but it has used older version of ignite and golang(v1.15). here is the link : https://github.com/cbonoz/cosmos20/blob/master/go.mod.

After installing dependencies I tried to build and run-- error app.js not found with some missing test case file. 
I tried with lower version of golang(v1.16) with the current version of ignite(v0.26.1) but this ignite version supports golang(v1.18 and above) 
I tried the older version of ignite (v0.25.2) because this version of ignite uses older version of golang(v1.16) and as the repo used the older version of golang. But the error for this version of ignite came---(THis VERSION OF IGNITE(v0.25.2) NO LONGER SUPPORTS COSMOS SDK)
1. Can someone help me out to either help me out to implement my problem in new version of ignite or either old version too. (i.e. WHAT ALL be needed if I want to use the module in my own project impemented in the above github repo I mentioned).

2. **Suggest some method that how can I create custom blockchain over COSMOS SDK that has option to fetch data from external apis**.

**Comment by MindlessSteel** [comment]
Maybe polygon will work for your foundation as far as the coding problem I have no idea 
Except maybe swift will work?

**Comment by facundomedica** [comment]
1. Regarding starting a new chain I suggest you take a look at our docs: https://docs.cosmos.network/v0.47 if you would like to use Ignite, please ask in their repo.
2. You can definitely do that with Cosmos SDK, depending on the type of data you'll need some kind of oracle module inside your chain (check Umee, Terra, there might be others that do this) and an external binary that fetches the API data turns it into a transaction and broadcasts it to your chain.

As a general suggestion, try going step by step and avoid using software that is too old, the SDK is constantly improving and code from 2 years go might not work as is (although you can make it work with some refactor).
